### DISCUSSION TITLE
async http request library

### DISCUSSION THREAD

**Discussion by Rai0Catodic0** [micropython/micropython#10247] [Ideas]
Hello everyone, 

I recently did a project using the ESP32 port, and I wanted to make async HTTP requisitions, but I think a library for that doesn't exist yet. I was thinking about trying to implement that, but I've never tried implementing a library for an embedded system. 
Do you guys have any tips on where to start and which parts of the repository should I look into? 

thanks : )

**Comment by git-n-pissed** [comment]
Maybe what you are talking about already exists?

https://github.com/pfalcon/picoweb

**Reply by Rai0Catodic0** [reply]
Thanks a lot for your answer. I just took a look at the link you send me, I understand that is a library that implements a async HTTP host, it's a framework for async web apps right? 
Im looking for a library for consume https APIs in a async manner. 

But your suggestions will be useful for some other projects I have in mind.

**Reply by peterhinch** [reply]
The preferred web framework is [microdot](https://microdot.readthedocs.io/en/latest/) rather than picoweb.

**Reply by git-n-pissed** [reply]
Been using picoweb for a while and had good results.  Why is microdot preferred?  Haven't done any analysis other than file size (microdot 41KB, picoweb 11KB).

**Reply by peterhinch** [reply]
Picoweb works fine, but there is a potential maintenance issue.

Picoweb is primarily targeted at Pycopy rather than MicroPython. It was written, and is maintained, by Paul Sokolovsky who is also the maintainer of Pycopy. In the past it required minor modification to run under MP. There is no guarantee of future MP compatibility.

**Reply by gambolputty** [reply]
@Rai0Catodic0 Did you find any solution? I'm looking for the same, but only find blocking HTTP clients.

**Reply by bulletmark** [reply]
There are two client libraries in micropython-lib: `aiohttp` and `uaiohttpclient` . Either can be installed using `mpremote mip install`.

I've used `uaiohttpclient` and it worked well for me.
